node/node_test.go

```diff
@@ -918,17 +918,26 @@ func TestNodeHybridTopology(t *testing.T) {
 
 	startAndConnectNodes(nodes, 10*time.Second)
 
+	// ensure the initial connectivity topology
+	require.Eventually(t, func() bool {
+		node0Conn := len(nodes[0].net.GetPeers(network.PeersConnectedIn)) > 0                             // has connection from 1
+		node1Conn := len(nodes[1].net.GetPeers(network.PeersConnectedOut, network.PeersConnectedIn)) == 2 // connected to 0 and 2
+		node2Conn := len(nodes[2].net.GetPeers(network.PeersConnectedOut, network.PeersConnectedIn)) >= 1 // connected to 1
+		return node0Conn && node1Conn && node2Conn
+	}, 60*time.Second, 500*time.Millisecond)
+
 	initialRound := nodes[0].ledger.NextRound()
 	targetRound := initialRound + 10
 
+	// ensure discovery of archival node by tracking its ledger
 	select {
 	case <-nodes[0].ledger.Wait(targetRound):
 		e0, err := nodes[0].ledger.Block(targetRound)
 		require.NoError(t, err)
 		e1, err := nodes[1].ledger.Block(targetRound)
```

Comment:
The issue could also be that one node just hasn't seen the latest block yet. Perhaps also wait for `node[1]` to reach the target round?

Reply:
The test checks node[0] can catchup from node[2] discovered via DHT. IMO there is no point of waiting for node[1] since it does not serve blocks to node[0]

Reply:
If there's no point in waiting for node[1], then why does the test ask node[1] for a block to compare it to node[0]'s response? Maybe it should ask node[2] for this instead?

Reply:
to check if two non-communicating nodes are on the same round. Does not matter much, can be removed

Reply:
The example runs you linked both fail on this line, so the nodes being out of sync seems like an issue. If it's not important to check this, removing it seems fine

Reply:
oh, you are right, I was working on some other example with "no block notification for wallet".
although it is still interesting why node[1] gets out of sync

Reply:
I was not able to repro and figure out why node 1 is behind so requiring equal stake for node 0 and node 1 to that they are always in sync.